jadx-core/src/main/java/jadx/core/dex/visitors/SimplifyVisitor.java

```diff
@@ -256,16 +257,12 @@ private static InsnNode simplifyArith(ArithNode arith) {
 			}
 			InsnArg firstArg = arith.getArg(0);
 			if (arith.getOp() == ArithOp.XOR && firstArg.getType() == ArgType.BOOLEAN) {
-				if (lit == 1 && firstArg.isInsnWrap()) {
-					InsnNode wr = ((InsnWrapArg) firstArg).getWrapInsn();
-					InsnNode not = new InsnNode(InsnType.NOT, 1);
-					not.setResult(wr.getResult());
-					not.addArg(firstArg);
-					return not; 
-				}
-				if (lit == 0 && firstArg.isInsnWrap()) {
-					return ((InsnWrapArg) firstArg).getWrapInsn();
-				}
+				InsnNode node = new InsnNode(lit == 0 ? InsnType.MOVE : InsnType.NOT, 1);
+				RegisterArg result = arith.getResult();
+				result.setType(ArgType.BOOLEAN);
```

Comment:
Don't change types, made checks for this before modifications.

Reply:
Actually, the type of the `firstArg` is checked, not `result`.

If this wasn't changed, then the type is `int`, and the code in `InsnGen.java` won't work:

    char op = insn.getResult().getType() == ArgType.BOOLEAN ? '!' : '~';